### PR TITLE
Making read file and read stdin data available through command class

### PR DESCRIPTION
**Pull request by jspruance**

### CODE REVIEW

File: packages/command/src/readpipeddata.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const ReadPipedStdin = {
```
[the remark this answers is not in the document]
Comment (jspruance) [reply packages/command/src/readpipeddata.ts]: unit test added

File: packages/command/src/readtextfile.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const fs = require('fs-extra');
```
[the remark this answers is not in the document]
Comment (jspruance) [reply packages/command/src/readtextfile.ts]: unit test added

File: packages/chatdown/src/commands/chatdown.ts [hunk 3]
```diff
@@ -74,16 +71,16 @@ export default class Chatdown extends Command {
...
     try {
-      // Check if path passed in --chat
+      // Check if path passed in --in
       if (args && args.length > 0) {
-        return txtfile.readSync(args)
+        return utils.readTextFile(args)
       } else {
         //Check if piped data was sent
         const {stdin} = process
         if (stdin.isTTY) {
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: Can we add this check to the utils as well?

File: packages/command/src/command.ts [hunk 4]
```diff
@@ -53,6 +52,10 @@ export abstract class Command extends Base {
     process.stdin.destroy()
   }
 
+  readStdin() {
+    return ReadPipedData.read()
```
Comment (munozemilio) [inline packages/command/src/command.ts]: Should we try catch this and return false in case no input is passed?

Comment (jspruance) [reply packages/command/src/command.ts]: Added

File: packages/chatdown/src/commands/chatdown.ts (hunk 3 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: This is done

**Pull request merged**